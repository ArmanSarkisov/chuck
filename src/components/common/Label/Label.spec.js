import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Label } from './index';

describe('Label test', () => {
  let component;

  describe('Has error text', () => {
    beforeEach(() => {
      component = shallow(
        <Label error="some error" htmlFor="text" text="cool" />,
      );
    });

    it('should render Label component with Error', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render props error', () => {
      const span = component.find('span');
      expect(span.text()).toBe('some error');
    });
  });

  describe('Has not error text', () => {
    beforeEach(() => {
      component = shallow(<Label htmlFor="text" text="cool" />);
    });

    it('should render Label component without Error', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render props text', () => {
      const span = component.find('span');
      expect(span.text()).toBe('cool');
    });
  });
});
