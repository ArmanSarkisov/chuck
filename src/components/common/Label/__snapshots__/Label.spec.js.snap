// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label test Has error text should render Label component with Error 1`] = `
<label
  className="block text-gray-700 text-sm font-bold mb-2 text-red-700"
  htmlFor="text"
>
  <span>
    some error
  </span>
</label>
`;

exports[`Label test Has not error text should render Label component without Error 1`] = `
<label
  className="block text-gray-700 text-sm font-bold mb-2"
  htmlFor="text"
>
  <span>
    cool
  </span>
</label>
`;
